### DISCUSSION TITLE
2-wire Start Mode

### DISCUSSION THREAD

**Discussion by MiddleSiggy** [jgyates/genmon#1201] [Ideas]
I have been a long time user in the "normal" configuration for a long while now, but I have recently put in a hybrid whole house solar system, and am now going to use my generator to only charge my batteries when there is no grid or solar power.

The way this happens is you put the controller in 2-wire mode (or enable 2-wire mode) and then the hybrid inverter, when it needs power, and there is no grid or solar it will start the generator remotely.

In this state a few things happen:
- I only have the T1 or 120v line installed feeding the 120v for the battery charger - curious if that is detectable? and if so can it be displayed on the screen in some way.
- I no longer have L1 & L2 from the grid feeding N1 & N2 so utility power is no longer tested or needed. (It would be nice to specify that I am running in the 2-wire mode and then no longer show the "utility voltage" because it is now going to be zero.
- - If T1 is available then maybe add T1 voltage instead
- - It might be possible to hook up N1 & N2 just to get this to display something, but not sure if that is really necessary
- Also when N1 & N2 are not installed:
- - On Status page: Line Utility Voltage, Max, Min Threshold Voltage really are not needed

If you would like more information on the 2-wire mode and my model please let me know I will share everything I can.

**Comment by jgyates** [comment]
Hi @MiddleSiggy 

The battery charger relay status is reportable via modbus but only on the liquid cooled units. Air cooled units do not export the charger relay status. There is some experimental data on the charger current displayed if you enable "Display Unknown Sensors" on the setting page. This is data that is present in the modbus registers but I do not know the exact puprose but it shows my current / best guess at what the data could be.

I can potentially add a two wire mode that will disable showing some info. If you want to send me your registers and logs I can do some testing. I don't know if T1 is exported over modbus but your register data may shed some light on that.

**Comment by interested1204** [comment]
I have recently installed a hybrid whole house solar system as well.  I think that MiddleSiggy is referencing using the Generac to charge the whole house batteries.  In my case, the generac was converted to 2 wire start mode then the Inverter (Sol-Ark 15k) determines when the whole house batteries are run down and fires up the generac to recharge them.   I've got mine "working" that way, though the 2 wire start doesn't show up as an "outage" so it doesn't calculate the fuel used for the outage.  Ironically, right now, its off but shows "Status: System in outage since 2025-05-18 16:01:18" when I was doing testing earlier and it still shows "System in Outage: Yes"

Would it be possible to have a software switch for 2 wire start usage where the system is activated by a switch in settings to treat 2 wire activations as outages so that it can calculate fuel usage etc?  (BTW, utility voltage is 0 so thats probably why it shows sys in outage/yes I'm guessing)

**Reply by jgyates** [reply]
@interested1204 

I would recommend setting the value "Do Not Check for Outages" to enable on the Settings page. The outage settings should not effect the power / fuel calculation. Of course your may or may not be able to use the power / fuel calculation depending on the controller type. 

If you think your controller is supported I would need more info to attempt to diagnose if you are not getting fuel calculation data. If you send your logs on the About page that would give me most of the data I would need (controller type, fuel type, do you have a power gauge on the web interface, etc).

**Reply by interested1204** [reply]
Oh I gotcha. that makes sense.     As far as usage, I was referencing the outage log has fuel usage per use.

    Outage Log : 
        04-18-2025 09:28:21, Duration: 0:01:24, Estimated Fuel: 0.01 gal
        04-18-2025 09:16:41, Duration: 0:00:25, Estimated Fuel: 0.00 gal
        04-17-2025 08:34:41, Duration: 7:53:11, Estimated Fuel: 1.40 gal
        04-06-2025 00:56:30, Duration: 0:00:06, Estimated Fuel: 0.03 gal
        04-06-2025 00:50:11, Duration: 0:00:21, Estimated Fuel: 0.12 gal

When selecting "Do Not Check for Outages" is there anywhere to see the per use fuel usage?   Love the software btw, appreciate all the hard word you've put into it.   I've talked 3 other people into using it, LOL.

Controller is an Evolution 2.0, Propane, power gauges of L1/L2/ Power Output all on the Status page.   Logs are on the way!

**Reply by jgyates** [reply]
No, the per use is only calculated for outages. The monthly totals of fuel usage are on the Maintenance page. 

I am glad it is working well for you and thanks for passing it along to others.
